generator/common/beatgen/setup/setup.go

```diff
@@ -0,0 +1,126 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package setup
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	devtools "github.com/elastic/beats/dev-tools/mage"
+	"github.com/magefile/mage/sh"
+	"github.com/pkg/errors"
+)
+
+// RunSetup runs any remaining setup commands after the vendor directory has been setup
+func RunSetup() error {
+	vendorPath := "./vendor/github.com/"
+
+	//Copy mage stuff
+	err := os.MkdirAll(filepath.Join(vendorPath, "magefile"), 0755)
+	if err != nil {
+		return errors.Wrap(err, "error making mage dir")
+	}
+
+	err = sh.Run("cp", "-R", filepath.Join(vendorPath, "elastic/beats/vendor/github.com/magefile/mage"), filepath.Join(vendorPath, "magefile"))
+	if err != nil {
+		return errors.Wrap(err, "error copying vendored magefile")
```

Comment:
Maybe errors.Wrap(err, "error copying vendored magefile from '%s' to '%s'")? :sweat_smile: